### PR TITLE
Update Management SDK for Microsoft.DigitalTwins to Support API Version 2022-05-31

### PR DESCRIPTION
**Pull request by efriesner**

### CODE REVIEW

File: sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure Digital Twins Management Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Digital Twins Resources.")]
 
-[assembly: AssemblyVersion("1.1.0.0")]
-[assembly: AssemblyFileVersion("1.1.0.0")]
+[assembly: AssemblyVersion("1.2.0.0")]
```
Comment (archerzz) [inline sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Properties/AssemblyInfo.cs:10]: You don't need to bump the `AsemblyVersion` if it's just a minor upgrade. See the [guide here](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=track-1-sdk-generation-from-updated-spec).

> AssemblyVersion: If it's a major version update (e.g. from 1.2.3.0 to 2.0.0.0), AssemblyVersion will also need to be changed

File: sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/IPrivateEndpointConnectionsOperations.cs [hunk 2]
```diff
@@ -136,7 +137,7 @@ public partial interface IPrivateEndpointConnectionsOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse<PrivateEndpointConnection>> CreateOrUpdateWithHttpMessagesAsync(string resourceGroupName, string resourceName, string privateEndpointConnectionName, PrivateEndpointConnectionProperties properties, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<AzureOperationResponse<PrivateEndpointConnection>> CreateOrUpdateWithHttpMessagesAsync(string resourceGroupName, string resourceName, string privateEndpointConnectionName, ConnectionProperties properties, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (archerzz) [inline sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/IPrivateEndpointConnectionsOperations.cs:140]: There are breaking changes in the interfaces. Here are 2 options:
  - Add manual codes to provide backward compatibility, see compute for reference: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations
- Or you need to bump the major version

File: sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/IPrivateEndpointConnectionsOperations.cs [hunk 3]
```diff
@@ -194,6 +196,6 @@ public partial interface IPrivateEndpointConnectionsOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse<PrivateEndpointConnection>> BeginCreateOrUpdateWithHttpMessagesAsync(string resourceGroupName, string resourceName, string privateEndpointConnectionName, PrivateEndpointConnectionProperties properties, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<AzureOperationResponse<PrivateEndpointConnection>> BeginCreateOrUpdateWithHttpMessagesAsync(string resourceGroupName, string resourceName, string privateEndpointConnectionName, ConnectionProperties properties, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (archerzz) [inline sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/IPrivateEndpointConnectionsOperations.cs:199]: Same

File: sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/Models/DigitalTwinsDescription.cs [hunk 4]
```diff
@@ -51,11 +53,13 @@ public DigitalTwinsDescription()
...
-            : base(location, id, name, type, tags, identity)
+        public DigitalTwinsDescription(string location, string id = default(string), string name = default(string), string type = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), DigitalTwinsIdentity identity = default(DigitalTwinsIdentity), SystemData systemData = default(SystemData), System.DateTime? createdTime = default(System.DateTime?), System.DateTime? lastUpdatedTime = default(System.DateTime?), string provisioningState = default(string), string hostName = default(string), IList<PrivateEndpointConnection> privateEndpointConnections = default(IList<PrivateEndpointConnection>), string publicNetworkAccess = default(string))
```
Comment (archerzz) [inline sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/Models/DigitalTwinsDescription.cs:61]: breaking change

File: sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/Models/GroupIdInformation.cs [hunk 5]
```diff
@@ -29,10 +29,11 @@ public GroupIdInformation()
         /// <summary>
         /// Initializes a new instance of the GroupIdInformation class.
         /// </summary>
+        /// <param name="properties">The group information properties.</param>
         /// <param name="id">The resource identifier.</param>
         /// <param name="name">The resource name.</param>
         /// <param name="type">The resource type.</param>
-        public GroupIdInformation(GroupIdInformationPropertiesModel properties, string id = default(string), string name = default(string), string type = default(string))
+        public GroupIdInformation(GroupIdInformationProperties properties, string id = default(string), string name = default(string), string type = default(string))
```
Comment (archerzz) [inline sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/Models/GroupIdInformation.cs:36]: Breaking change

File: sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/Models/PrivateEndpointConnection.cs [hunk 6]
```diff
@@ -31,15 +31,19 @@ public PrivateEndpointConnection()
...
         /// Initializes a new instance of the PrivateEndpointConnection class.
         /// </summary>
+        /// <param name="properties">The connection properties.</param>
         /// <param name="id">The resource identifier.</param>
         /// <param name="name">The resource name.</param>
         /// <param name="type">The resource type.</param>
-        public PrivateEndpointConnection(PrivateEndpointConnectionProperties properties, string id = default(string), string name = default(string), string type = default(string))
+        /// <param name="systemData">Metadata pertaining to creation and last
+        /// modification of the private endpoint connection.</param>
+        public PrivateEndpointConnection(ConnectionProperties properties, string id = default(string), string name = default(string), string type = default(string), SystemData systemData = default(SystemData))
```
Comment (archerzz) [inline sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/Models/PrivateEndpointConnection.cs:40]: breaking change

File: sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/PrivateEndpointConnectionsOperationsExtensions.cs [hunk 7]
```diff
@@ -166,8 +166,9 @@ public static async Task DeleteAsync(this IPrivateEndpointConnectionsOperations
             /// The name of the private endpoint connection.
             /// </param>
             /// <param name='properties'>
+            /// The connection properties.
             /// </param>
-            public static PrivateEndpointConnection CreateOrUpdate(this IPrivateEndpointConnectionsOperations operations, string resourceGroupName, string resourceName, string privateEndpointConnectionName, PrivateEndpointConnectionProperties properties)
```
Comment (archerzz) [inline sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/PrivateEndpointConnectionsOperationsExtensions.cs:170]: breaking change

**Review by archerzz — CHANGES_REQUESTED** [review]
There are breaking changes most of which seems model renaming. Consider provide backward compatibility customization, or you need to bump the major version (e.g. `2.0.0.0`).

File: sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/IPrivateEndpointConnectionsOperations.cs (hunk 2 above)
Comment (efriesner) [reply sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/IPrivateEndpointConnectionsOperations.cs:140]: This is why we added the models in the Customized/Models folder, based on comments from the PR of the preview version.

File: sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (efriesner) [reply sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Properties/AssemblyInfo.cs]: Reverted this file

**Comment by sjiherzig** [comment]
@archerzz Do the customized models cover the breaking changes you identified? If so, can you please approve this PR?

File: sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/IPrivateEndpointConnectionsOperations.cs (hunk 3 above)
Comment (efriesner) [reply sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/IPrivateEndpointConnectionsOperations.cs:199]: Add new interface in Customized folder for backwards compatibility

File: sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/PrivateEndpointConnectionsOperations.cs [hunk 8]
```diff
@@ -471,14 +479,15 @@ public async Task<AzureOperationResponse> DeleteWithHttpMessagesAsync(string res
...
+        /// The connection properties.
         /// </param>
         /// <param name='customHeaders'>
         /// The headers that will be added to request.
         /// </param>
         /// <param name='cancellationToken'>
         /// The cancellation token.
         /// </param>
-        public async Task<AzureOperationResponse<PrivateEndpointConnection>> CreateOrUpdateWithHttpMessagesAsync(string resourceGroupName, string resourceName, string privateEndpointConnectionName, PrivateEndpointConnectionProperties properties, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse<PrivateEndpointConnection>> CreateOrUpdateWithHttpMessagesAsync(string resourceGroupName, string resourceName, string privateEndpointConnectionName, ConnectionProperties properties, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
[the remark this answers is not in the document]
Comment (efriesner) [reply sdk/digitaltwins/Microsoft.Azure.Management.DigitalTwins/src/Generated/PrivateEndpointConnectionsOperations.cs:490]: @archerzz I have added the corresponding APIs to the Customized folder

**Review by archerzz — APPROVED** [review]

**Pull request merged**